### PR TITLE
Remove usage of Maven SCM API

### PR DESCRIPTION
**Pull request by basil**
### Problem

We currently rely on an old version of the Maven SCM API, which is not only inefficient but which also is a maintenance liability as can be seen in https://github.com/jenkinsci/plugin-compat-tester/pull/393#pullrequestreview-1245652252.

### Solution

Switch to running Git commands directly. Simpler and easier to maintain, and more efficient to boot.

### Implementation

Extracted and cleaned up from #382 by @olamy (credited as `Co-authored-by` in the commit message), but without two portions of that PR:

- The unnecessary cleanup changes have been removed. These belong in a separate code cleanup PR.
- The critical change to switch from deep to shallow cloning has been removed. This is not because I do not think it is desirable, but because it introduces risk and is not related to the core purpose of this PR. Once this PR is merged, I fully hope and expect for the change from deep to shallow cloning to proceed in #382. That will become only a few lines of code change once this PR is merged.

### Testing done

Tested in context in `jenkinsci/bom`. I also manually corrupted the Git URL to make sure the error handling code path was exercised and worked as expected.

### CODE REVIEW

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 1]
```diff
@@ -620,30 +609,122 @@ public void cloneFromSCM(PomData pomData, String name, String version, File chec
...
+
+        // git init
+        Process p = new ProcessBuilder()
+                .directory(checkoutDirectory)
+                .command("git", "init")
+                .redirectErrorStream(true)
+                .start();
+        try {
+            int exitStatus = p.waitFor();
+            String output = new String(p.getInputStream().readAllBytes(), Charset.defaultCharset()).trim();
```
[the remark this answers is not in the document]
Comment (basil) [reply src/main/java/org/jenkins/tools/test/PluginCompatTester.java]: Actually I did it in c20578d76dd6ca6c0c2095fc121a541223e319fe; some slight duplication with my other PR, but that will collapse based on which PR is merged first

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 2]
```diff
@@ -620,30 +610,131 @@ public void cloneFromSCM(PomData pomData, String name, String version, File chec
...
+     * @param connectionURL The connection URL, in a format such as
+     *     scm:git:https://github.com/jenkinsci/mailer-plugin.git or
+     *     https://github.com/jenkinsci/mailer-plugin.git
+     * @param scmTag the tag or sha1 hash to clone
+     * @param checkoutDirectory the directory in which to clone the Git repository
+     * @throws IOException if an error occurs
+     */
+    public static void clone(String connectionURL, String scmTag, File checkoutDirectory)
+            throws IOException {
+        System.out.println("Checking out from SCM connection URL " + connectionURL + "  at " + scmTag);
```
[the remark this answers is not in the document]
Comment (basil) [reply src/main/java/org/jenkins/tools/test/PluginCompatTester.java:654]: As is obvious and did not need to be stated.

**Comment by jtnord** [comment]
note: needs local adaptions, I am working on updating.

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 3]
```diff
@@ -620,30 +610,131 @@ public void cloneFromSCM(PomData pomData, String name, String version, File chec
...
+                throw new IOException("git init failed with exit status " + exitStatus + ": " + output);
+            }
+        } catch (InterruptedException e) {
+            throw new IOException("git init was interrupted", e);
+        }
+
+        // git fetch ${CONNECTION_URL} ${SCM_TAG}
+        String gitUrl;
+        if (StringUtils.startsWith(connectionURL, "scm:git:")) {
+            gitUrl = StringUtils.substringAfter(connectionURL, "scm:git:");
```
[the remark this answers is not in the document]
Comment (jtnord) [reply src/main/java/org/jenkins/tools/test/PluginCompatTester.java:692]: fixed the typo above.

**Comment by basil** [comment]
How's this looking this week James?

**Comment by jtnord** [comment]
> How's this looking this week James?

the prerequisite for testing this was just merged, just updating the PR to check now.

**Review by jtnord — APPROVED** [review]
tested, all ok.

**Comment by basil** [comment]
Awesome, thank you!

**Pull request merged**